keps/sig-auth/4940-psa-block-host-field-in-probes/README.md

```diff
@@ -0,0 +1,700 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+We have a `Host` field that can be set from `TCPSocketAction` and
+`HTTPGetAction` fields which are part of the `ProbeHandler` and
+`LifecycleHandler` structs in Kubernetes. The `Host` field is used
+for allowing users to specify another entity other than the
+podIP (which is the default value) to which Kubelet should
+perform probes to. However this opens it up for security attacks
+since the `Host` field can be set to pretty much any value in the
+system including security sensitive external hosts or localhost on
+the node. Kubelet will be probing this set `Host` value which can
+lead to blind SSRF attacks.
+
+## Motivation
+
+**Probe-based Blind SSRF Attacks**: The current definition of TCP and
+HTTP probes allows the user to specify an alternative hostname/IP to
+connect to rather than the pod IP. (The expected use is for sending a
+probe via a HostPort, NodePort, or LoadBalancer IP.) But this allows a
+"blind SSRF" (Server-Side Request Forgery) attack, in which a pod can
+trick kubelet into sending an HTTP GET request to an arbitrary URL (or
+portscanning TCP ports on arbitrary hosts). ([kubernetes #99425])
+
+## Proposal
+
+There is a long term plan to deprecate the existing TCP and HTTP probe
+types in the API to replace them with ones with slightly different semantics.
+See the [KEP](https://github.com/kubernetes/enhancements/pull/4558) for more
+details. Given the unsolvable security problems with the Host field,
+we do not plan to offer it in the new types.
+
+Meanwhile, the older API is never going to go away. So we also want to
+add PSA to allow admins to be able to restrict users from creating
+probes with the Host field set when using the (about to be deprecated) API.
+Here is the [draft PR](https://github.com/kubernetes/kubernetes/pull/125271)
```

Comment:
(again I'd mention the PR number in the text. "This is implemented by [kubernetes #125271]" or something)